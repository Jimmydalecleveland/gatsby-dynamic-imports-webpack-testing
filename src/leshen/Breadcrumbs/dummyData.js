export const CRUMB_DATA = [
  {
    shortName: 'Home',
    path: '/',
  },
  {
    shortName: 'Internet',
    path: '/internet',
  },
  {
    shortName: 'Gigabit',
    path: '/gigabit',
  },
  {
    shortName: 'Gigabit Plans',
    path: '/gigabit-plans',
  },
]
