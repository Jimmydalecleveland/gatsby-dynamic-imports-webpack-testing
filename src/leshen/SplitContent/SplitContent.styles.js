import styled from "@emotion/styled"
import { css } from "@emotion/core"
import { stripUnit } from "polished"

import SectionComponent from "../Section"
import fluidRange from "../utils/fluidRange"

export const Section = styled(SectionComponent)(
  ({
    theme,
    imageWidth,
    alignImageToBottom,
    reverseForMobile,
    hideImageOnMobile,
  }) => {
    const oneThird = stripUnit(theme.breakpoints.xl) / 3
    const oneFourth = stripUnit(theme.breakpoints.xl) / 4
    const width = stripUnit(imageWidth) || stripUnit(theme.breakpoints.xl)

    const gridTemplateColumns = ["1fr", "1fr"]
    const gridTemplateAreas = ["image", "content"]
    const reverseMobileTemplateAreas = alignImageToBottom
      ? true
      : reverseForMobile
    let mobileTemplateAreas = '"image" "content"'

    if (reverseMobileTemplateAreas) {
      mobileTemplateAreas = '"content" "image"'
    }

    /**
     * Determine if the image should be smaller than half the container
     *  and set the template areas to position the image on the correct side
     */
    if (width < oneFourth) {
      gridTemplateAreas.push("content")
      gridTemplateColumns.push("1fr")
    }
    if (width < oneThird) {
      gridTemplateAreas.push("content")
      gridTemplateColumns.push("1fr")
    }

    return css`
      ${alignImageToBottom && `padding-bottom: 0 !important`};

      &:nth-of-type(odd) {
        .split-content__grid {
          @media screen and (min-width: ${theme.breakpoints.md}) {
            grid-template-areas: "${[...gridTemplateAreas]
              .reverse()
              .join(" ")}";
          }
        }
      }

      .split-content {
        &__grid {
          display: grid;
          align-items: center;
          grid-template-rows: auto auto;
          grid-template-areas: ${mobileTemplateAreas};
          ${fluidRange("grid-gap", [
            stripUnit(theme.spacing.lg),
            stripUnit(theme.spacing.xxl),
          ])};

          @media screen and (min-width: ${theme.breakpoints.md}) {
            grid-template-areas: "${gridTemplateAreas.join(" ")}";
            grid-template-columns: ${gridTemplateColumns.join(" ")};
            grid-template-rows: none;
          }
        }

        &__image {
          align-self: ${alignImageToBottom ? "flex-end" : "center"};
          margin: 0;
          padding: 0;
          text-align: center;
          ${hideImageOnMobile ? "display: none" : null};
          grid-area: image;

          @media screen and (min-width: ${theme.breakpoints.md}) {
            display: block;
          }
        }

        &__content {
          display: flex;
          flex-direction: column;
          align-items: flex-start;
          grid-area: content;
          ${
            alignImageToBottom &&
            fluidRange("padding-bottom", [
              stripUnit(theme.spacing.lg),
              stripUnit(theme.spacing.xxl),
            ])
          };
        }
      }
    `
  }
)
