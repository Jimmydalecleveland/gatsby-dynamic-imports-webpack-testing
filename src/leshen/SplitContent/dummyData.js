import lifestyle from '../../images/split-content-lifestyle.jpg'
import repRyan from '../../images/rep-ryan.png'

export const FLUID_IMAGE_DATA = {
  context: {
    custom: {
      alt: 'Alt Text',
    },
  },
  fluid: {
    base64:
      'data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAPABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAQFAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAABAv/aAAwDAQACEAMQAAABY4VxKBIGf//EABoQAAIDAQEAAAAAAAAAAAAAAAABAgMREjL/2gAIAQEAAQUCrjpP1qE3lksOz//EABYRAQEBAAAAAAAAAAAAAAAAAAAREv/aAAgBAwEBPwHSv//EABURAQEAAAAAAAAAAAAAAAAAAAAR/9oACAECAQE/AYj/xAAcEAACAQUBAAAAAAAAAAAAAAAAATECEBJBUWH/2gAIAQEABj8C6JUySZ6F7b//xAAcEAACAgIDAAAAAAAAAAAAAAABEQAxIVEQQZH/2gAIAQEAAT8hIHQaigsQF4gXT2NWQVuEdhW3x//aAAwDAQACAAMAAAAQ4+//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EMVUo//EABYRAQEBAAAAAAAAAAAAAAAAAAARUf/aAAgBAgEBPxDaH//EAB4QAQACAgEFAAAAAAAAAAAAAAERIQAxcUFRYZGx/9oACAEBAAE/EEKI60I86xyRZUEjqPvNJHwy9miG2LXgZRab+9sUtAHGf//Z',
    src: lifestyle,
    srcSet: lifestyle,
    aspectRatio: 1.3524904214559388,
    sizes: '(max-width: 706px) 100vw, 706px',
    presentationWidth: 2560,
  },
}

export const FLUID_IMAGE_DATA_BOTTOM = {
  context: {
    custom: {
      alt: 'Meet Ryan',
    },
  },
  fluid: {
    base64:
      'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAjCAYAAACU9ioYAAAACXBIWXMAAAsSAAALEgHS3X78AAAIKklEQVRIx42WeVQVRxbGWxnnTOKZ5MQQ9CjhmNFk9IhGE0Ej4B4St1EREomCqCAoIiDCKLsDAi6ARAiL4IJRSUhiBETGfYm7EkVke2xv4+2v3751d92p6ockf432OfWqu2/1r+/9vqp6TVF/OgCAGjFs+DByvsDH7/1xbm7V77391omJHu5Lyb03yM9HsVTi1o3Uax3zZs/mYUv8l42cPX1aq7/PLFi3wh82rPoC/H1neZFYgP+84a8FMyFEbQxczg9eOGeW93eZ8SC+28gqmm9Zhbd+haz4iGvOKkzDXjvDAH8/F9LP9Z6+pSQtBhorC9kHv57hyjN3wb6ESI2Pr68biUeuCxr2SpgR67dsgY+Ls/RPA2PXrYKMiGDuv2W5qCIlhq0r3Q8Xjh3ZQeLHC7JcXl2yHai0iOC/kPM9YYGZZw6mwYOzZY7yPdvg5/xM5vbpUnh0oSaHxK/VVr0a6OgAKidmI69h3o5Nyw8mRkF2dCjK2rEVFSUnsHsivoH4yE2829GhX7s4HNz/B7IsS20Pj6DGjHqbh/rNnJG9apEvnMvfy1wpK4SVC+fSHh7/GE9iRwoLXYxGKyXsFvNTTa83/wGy2eyU1eoYuq49dWIE6YMXe3t+OnkizPnYkwlYOBe2hwTQYGtzHXrObqEMRquLXm/hrw0GC6XVGKkhmMViH47bkDa97a3HqnISYLnvx2zEmsXoek0xKCXCJg4gWKfVjftzdVqtkXddqdQ5b9C0cWgasAw3zWy2NRoNVnh64zy69WMJenD+KDRfqkGswwHksJjtelzyHb3BkmS1M/wC0mgGGThVXjO7nfWyWuxXcc+Sh2jazKmUNJJL5dBy/Tzqa7kPVhvL2GwMi8fAy4OmTQ0vl61KpXdmJRYrRplMNjkZgAcju51hMBxZzDawOxDSKhRIq6aRyWhFGIibA1ksDhbLZSPPKOR0DOGolDqnZDKZJnMQZsWDEMkATwvE4IYlAJbjY+Qe7hkeajbbAZfN2mwcyGXaXqlE+YYzO5H8TYWCFlrMDsADOWwMcjhYxGAQBvIQjkNArnGcAHm4XmdGapUeYUNYrcYEErHiSx7Y0y3yVMi1OG0NDEhUZAByPujMCMPhJVxnNIGgvw/a2zvRixcCEPXLkVZjYGxWBAqloZQHSsWyIPw2wAHu6ZM21NbaTaCAS8I62Uk/2GyAdQYtbQSsObx4LkDCPjkyW1jObGagv7u/62b92RGU1cYlE/1sds4hl2nQnZuP0PNnXSASyhGeAWDCIPNgM5qsoFEboL93AHV3ieHejbvo6W/X2eZrdehSdQH8VJI1n7pZX5MrFbQCrVY7RH1iaH74HHq7JUgqVgLWiAdho5xAoxMoG9CwGpUR7t+8c7n224yW2z+VQWN5JhzPid9LVWbHVN07Vwatv11gr9R+b3l8r5nBAoNSQSOdzsRDTCZnI+c6LI9aZWB02AipVJtSlbUrtbEyBxrKM+BYTtwV6mh2bBMBNl/+Aa7UVN35/eEztWxADQNSNa8l0e2lfgRKgFKJilEpdCAVy5MXurmMLM3YTl+uzoUzhbs7qMp9sY8f1lfA9bNFUF91OPH+zbu9SjlNJiteKXrAOwkQLUnDGwG5TzRk1DjW3yOMI8ZW56fWXD11AE4X7Kap0/lJwqeXqqGhKld7+Yfjnzy591gs6pcRnTg8YUl5ZBmSJQbkBQMSNZ5zSkZLzBkE/liSXdp4NBuK06NZ6ruMrZpH9eVwr6G6ta+z26e9pcMsFsrJQwiXxgO1WgLTOYFYCgLUYMOkIlk6AZ48uOdq7eFkyIkPpanUyCC66dg+uFFbKu5pa/fpFQi1EpGCzxAbwztNysS7CTZKx5/39UgZWk1eQucTYEXWzs6TufGQsOFfAipp08rmU/tj4eKJA/BzxQEvQUfP7wa9lWTEkQxJiXif40EKGQ14R0FikYIx6m04ptzM7+Cp0cKytCgIX73gMhUXsvSXY/u2w/nSTKjMSVh08VxDc8uTVpyFhMNrHMiyJJmpsWYqlQ66OvpR86PnjKhvAER90m2Y99dDieHKgztDIWT53AQq6mv/lOLUzXAyNxYq8naHdndJesRCBTxrbuPw8uIn8oBUBf0Y0NHWC60tAsCrBJtiImYdmj72zfH748NgZ8gKm7/PjAlUWIC/d+a2r6A0bQscP5xbLugU6RQ4o4EBDb8EnzxsJQbxQIkYm6Q2om6BhFHKsElqU1l25JrPjv8nDiID/esGPxe6qB3rlj0u3r0RasoKmgSdYjMBSLGbKuywCGfb/qKXh5EXdXWKoEcgYfW0FeRyXVNtYVrlSQyMWvPF9qE/mZi1SzOP4gwvVJfoMICzmMjeyOEdmwGHHW+uNgTEKKVchyUwEhmQVKqBR9euMvUl+6AMf7ZkRIf5DQH/vSnQryQ5Cm6cPQJHj5Sh9PQiVFlZC9XVv6AzZ85DXd1VdLHxNqqqqoW9e4tQVFQ6BKzeigp3xUBdUTZXkb4TsuKivhwCxm1c716QtMVw4/S3cK44F/1zwmcw2XMFmjJ1OXJ3nw1uo6eh0aM9kavrVHjnHU801t0HLfZdBZnha9Gh6HA2b2sYxK4LmjcE3LU59P28uHDdxfJD+DsmFUWvDYbpM4PAyzsIJnw4H8OmwHuuH4Cr6yTk4eEFk6csgSXzV8K2gGVscUIkJK0PlH7+iefIIWB0aMjfkkKCXlSlxEJe9AY2ZUMQfL5oPUydsRqNHTcTRr07Ef39LXf0ruskGDNmGvrwo/nog/EzudXz50Bi6Fd38qLDJhFOSmjQsP8BqOijO8GjUN8AAAAASUVORK5CYII=',
    src: repRyan,
    srcSet: repRyan,
    aspectRatio: 0.5787545787545788,
    sizes: '(max-width: 316px) 100vw, 316px',
    presentationWidth: 316,
  },
}
