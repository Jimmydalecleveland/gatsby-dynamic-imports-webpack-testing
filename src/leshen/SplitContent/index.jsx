import React from 'react'
import PropTypes from 'prop-types'
import _get from 'lodash.get'

import Img from '../Image'
import Typography from '../Typography'
import Placeholder from '../Placeholder'
import Breadcrumbs from '../Breadcrumbs'
import * as Styled from './SplitContent.styles'

const SplitContent = ({
  alignImageToBottom,
  reverseForMobile,
  hideImageOnMobile,
  image,
  heading,
  subheading,
  mainContent,
  children,
  breadcrumbProps,
  backgroundColor,
  headingLevel,
  isImageCritical,
  ...props
}) => {
  const imageType = _get(image, 'svg.presentationWidth') ? 'svg' : 'fluid'
  const imageWidth = _get(image, `${imageType}.presentationWidth`)

  // TODO: [LC-314] remove className from root element and use data-attribute
  return (
    <Styled.Section
      hideImageOnMobile={hideImageOnMobile}
      alignImageToBottom={alignImageToBottom}
      imageWidth={imageWidth}
      reverseForMobile={reverseForMobile}
      backgroundColor={backgroundColor}
      {...props}
    >
      {breadcrumbProps && (
        <Breadcrumbs {...breadcrumbProps} backgroundColor={backgroundColor} />
      )}
      <div className="split-content__grid">
        <div className="split-content__image">
          {image && Object.keys(image).length ? (
            <Img data={image} isImageCritical={isImageCritical} />
          ) : (
            <Placeholder type="error">
              <Typography variant="small">Notice</Typography>
              <Typography>
                The content entry of type &quot;SplitContent&quot; is missing
                the Image media asset.
              </Typography>
            </Placeholder>
          )}
        </div>
        <div className="split-content__content">
          <Typography variant={`h${headingLevel}`}>{heading}</Typography>
          {subheading && (
            <Typography
              variant={`h${headingLevel + 3}`}
              rendersAs={`h${headingLevel + 1}`}
            >
              {subheading}
            </Typography>
          )}
          {mainContent}
          {children}
        </div>
      </div>
    </Styled.Section>
  )
}

SplitContent.defaultProps = {
  alignImageToBottom: false,
  hideImageOnMobile: false,
  reverseForMobile: false,
  subheading: null,
  headingLevel: 2,
  image: null,
  mainContent: null,
  children: null,
  backgroundColor: null,
  breadcrumbProps: null,
  isImageCritical: false,
}

SplitContent.propTypes = {
  alignImageToBottom: PropTypes.bool,
  image: PropTypes.shape({
    fluid: PropTypes.object,
    fixed: PropTypes.object,
    svg: PropTypes.object,
    context: PropTypes.object,
  }),
  hideImageOnMobile: PropTypes.bool,
  reverseForMobile: PropTypes.bool,
  heading: PropTypes.string.isRequired,
  subheading: PropTypes.string,
  headingLevel: PropTypes.oneOf([1, 2]),
  mainContent: PropTypes.node,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.objectOf(PropTypes.node),
    PropTypes.arrayOf(PropTypes.node),
  ]),
  backgroundColor: PropTypes.string,
  breadcrumbProps: PropTypes.shape({
    crumbs: PropTypes.string,
    path: PropTypes.string,
    siteURL: PropTypes.string,
  }),
  isImageCritical: PropTypes.bool,
}

export default SplitContent

export const ticketID = 'CMS-176'
