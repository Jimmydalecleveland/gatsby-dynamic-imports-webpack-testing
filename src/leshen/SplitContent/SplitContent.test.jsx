import React from 'react'
import { render } from 'setup-tests'

import SplitContent from '.'
import { FLUID_IMAGE_DATA } from './dummyData'

describe('SplitContent', () => {
  test('should render', () => {
    const { container } = render(
      <SplitContent
        image={FLUID_IMAGE_DATA}
        heading="Heading"
        subheading="Subheading"
        mainContent={
          <>
            <p>Main Content</p>
          </>
        }
      />
    )
    expect(container).toMatchSnapshot()
  })

  test('should not throw an error without an image', () => {
    /**
     * SplitContent uses our 'Image' component, which expects ImageSharp data.
     * However, our build should not break because someone did not pass the correct data.
     * The result of this call should return an error placeholder component unless in production,
     * where it would return null (Placeholder logic)
     */
    const { container } = render(
      <SplitContent
        image={{}}
        heading="Heading"
        subheading="Subheading"
        mainContent={
          <>
            <p>Main Content</p>
          </>
        }
      />
    )
    expect(container).toMatchSnapshot()
  })

  test('should align image to bottom', () => {
    const { container } = render(
      <SplitContent
        image={FLUID_IMAGE_DATA}
        heading="Heading"
        subheading="Subheading"
        alignImageToBottom
        mainContent={
          <>
            <p>Main Content</p>
          </>
        }
      />
    )
    expect(container.firstChild).toHaveStyle('padding-bottom: 0')
    expect(container.firstChild.firstChild.firstChild).toHaveStyle(
      'align-self: flex-end'
    )
  })
})
