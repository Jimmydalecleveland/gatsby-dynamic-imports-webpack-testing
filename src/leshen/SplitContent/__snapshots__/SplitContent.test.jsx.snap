// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitContent should not throw an error without an image 1`] = `
.emotion-15 {
  padding-left: calc(2.2222222222222223vw + 21.333333333333332px);
  padding-right: calc(2.2222222222222223vw + 21.333333333333332px);
  padding-top: calc(7.777777777777778vw + -5.333333333333333px);
  padding-bottom: calc(7.777777777777778vw + -5.333333333333333px);
  position: relative;
}

@media (max-width:480px) {
  .emotion-15 {
    padding-left: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-15 {
    padding-left: 48px;
  }
}

@media (max-width:480px) {
  .emotion-15 {
    padding-right: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-15 {
    padding-right: 48px;
  }
}

@media (min-width:1288px) {
  .emotion-15 {
    padding-right: calc((100% - 1200px) / 2);
    padding-left: calc((100% - 1200px) / 2);
  }
}

@media (max-width:480px) {
  .emotion-15 {
    padding-top: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-15 {
    padding-top: 88px;
  }
}

@media (max-width:480px) {
  .emotion-15 {
    padding-bottom: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-15 {
    padding-bottom: 88px;
  }
}

@media screen and (min-width:768px) {
  .emotion-15:nth-of-type(odd) .split-content__grid {
    grid-template-areas: "content image";
  }
}

.emotion-15 .split-content__grid {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: auto auto;
  grid-template-areas: "image" "content";
  grid-gap: calc(4.444444444444445vw + 10.666666666666666px);
}

@media (max-width:480px) {
  .emotion-15 .split-content__grid {
    grid-gap: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-15 .split-content__grid {
    grid-gap: 64px;
  }
}

@media screen and (min-width:768px) {
  .emotion-15 .split-content__grid {
    grid-template-areas: "image content";
    grid-template-columns: 1fr 1fr;
    grid-template-rows: none;
  }
}

.emotion-15 .split-content__image {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 0;
  padding: 0;
  text-align: center;
  grid-area: image;
}

@media screen and (min-width:768px) {
  .emotion-15 .split-content__image {
    display: block;
  }
}

.emotion-15 .split-content__content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  grid-area: content;
}

.emotion-10 {
  font-weight: 800;
  font-size: calc(2.7777777777777777vw + 14.666666666666666px);
  line-height: calc(3.3333333333333335vw + 16px);
}

@media (max-width:480px) {
  .emotion-10 {
    font-size: 28px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    font-size: 48px;
  }
}

@media (max-width:480px) {
  .emotion-10 {
    line-height: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-10 {
    line-height: 56px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-10 {
    width: 100%;
  }
}

.emotion-12 {
  font-weight: 500;
  font-size: calc(0.5555555555555556vw + 17.333333333333332px);
  line-height: calc(1.1111111111111112vw + 18.666666666666668px);
}

@media (max-width:480px) {
  .emotion-12 {
    font-size: 20px;
  }
}

@media (min-width:1200px) {
  .emotion-12 {
    font-size: 24px;
  }
}

@media (max-width:480px) {
  .emotion-12 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-12 {
    line-height: 32px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-12 {
    width: 100%;
  }
}

.emotion-8 {
  background-color: #f4f5f6;
  border-radius: 4px;
  border: 1px solid #ff3068;
  box-shadow: 0 0.3rem 1rem 0 rgba(0,0,0,0.1);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-8 * {
  color: #383d43;
}

.emotion-0 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  background-color: #ff3068;
  padding: 24px 16px;
}

.emotion-0 svg {
  fill: #000;
}

.emotion-6 {
  color: #383d43;
  padding: 12px 16px;
  text-align: left;
}

.emotion-6 p {
  margin: 0;
}

.emotion-2 {
  font-size: 0.875rem;
  line-height: 1.375rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    width: 100%;
  }
}

.emotion-4 {
  font-size: 1rem;
  line-height: 1.375rem;
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-4 {
    width: 100%;
  }
}

<div>
  <section
    class="emotion-14 emotion-15 emotion-16"
  >
    <div
      class="split-content__grid"
    >
      <div
        class="split-content__image"
      >
        <div
          class="emotion-8 emotion-9"
          color="error"
        >
          <div
            class="emotion-0 emotion-1"
            color="error"
          >
            <span>
              Error icon Mock
            </span>
          </div>
          <div
            class="emotion-6 emotion-7"
          >
            <p
              class="emotion-2 emotion-3"
            >
              Notice
            </p>
            <p
              class="emotion-4 emotion-3"
            >
              The content entry of type "SplitContent" is missing the Image media asset.
            </p>
          </div>
        </div>
      </div>
      <div
        class="split-content__content"
      >
        <h2
          class="emotion-10 emotion-11"
        >
          Heading
        </h2>
        <h3
          class="emotion-12 emotion-13"
        >
          Subheading
        </h3>
        <p>
          Main Content
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`SplitContent should render 1`] = `
.emotion-7 {
  padding-left: calc(2.2222222222222223vw + 21.333333333333332px);
  padding-right: calc(2.2222222222222223vw + 21.333333333333332px);
  padding-top: calc(7.777777777777778vw + -5.333333333333333px);
  padding-bottom: calc(7.777777777777778vw + -5.333333333333333px);
  position: relative;
}

@media (max-width:480px) {
  .emotion-7 {
    padding-left: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-7 {
    padding-left: 48px;
  }
}

@media (max-width:480px) {
  .emotion-7 {
    padding-right: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-7 {
    padding-right: 48px;
  }
}

@media (min-width:1288px) {
  .emotion-7 {
    padding-right: calc((100% - 1200px) / 2);
    padding-left: calc((100% - 1200px) / 2);
  }
}

@media (max-width:480px) {
  .emotion-7 {
    padding-top: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-7 {
    padding-top: 88px;
  }
}

@media (max-width:480px) {
  .emotion-7 {
    padding-bottom: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-7 {
    padding-bottom: 88px;
  }
}

@media screen and (min-width:768px) {
  .emotion-7:nth-of-type(odd) .split-content__grid {
    grid-template-areas: "content image";
  }
}

.emotion-7 .split-content__grid {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-rows: auto auto;
  grid-template-areas: "image" "content";
  grid-gap: calc(4.444444444444445vw + 10.666666666666666px);
}

@media (max-width:480px) {
  .emotion-7 .split-content__grid {
    grid-gap: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-7 .split-content__grid {
    grid-gap: 64px;
  }
}

@media screen and (min-width:768px) {
  .emotion-7 .split-content__grid {
    grid-template-areas: "image content";
    grid-template-columns: 1fr 1fr;
    grid-template-rows: none;
  }
}

.emotion-7 .split-content__image {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 0;
  padding: 0;
  text-align: center;
  grid-area: image;
}

@media screen and (min-width:768px) {
  .emotion-7 .split-content__image {
    display: block;
  }
}

.emotion-7 .split-content__content {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  grid-area: content;
}

.emotion-0 {
  max-width: 2560px;
  width: 100%;
  margin: 0 auto;
}

.emotion-2 {
  font-weight: 800;
  font-size: calc(2.7777777777777777vw + 14.666666666666666px);
  line-height: calc(3.3333333333333335vw + 16px);
}

@media (max-width:480px) {
  .emotion-2 {
    font-size: 28px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    font-size: 48px;
  }
}

@media (max-width:480px) {
  .emotion-2 {
    line-height: 32px;
  }
}

@media (min-width:1200px) {
  .emotion-2 {
    line-height: 56px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-2 {
    width: 100%;
  }
}

.emotion-4 {
  font-weight: 500;
  font-size: calc(0.5555555555555556vw + 17.333333333333332px);
  line-height: calc(1.1111111111111112vw + 18.666666666666668px);
}

@media (max-width:480px) {
  .emotion-4 {
    font-size: 20px;
  }
}

@media (min-width:1200px) {
  .emotion-4 {
    font-size: 24px;
  }
}

@media (max-width:480px) {
  .emotion-4 {
    line-height: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-4 {
    line-height: 32px;
  }
}

@media all and (-ms-high-contrast:none),(-ms-high-contrast:active) {
  .emotion-4 {
    width: 100%;
  }
}

<div>
  <section
    class="emotion-6 emotion-7 emotion-8"
  >
    <div
      class="split-content__grid"
    >
      <div
        class="split-content__image"
      >
        <div
          class="emotion-0 emotion-1 gatsby-image-wrapper"
          style="position: relative; overflow: hidden;"
        >
          <div
            aria-hidden="true"
            style="width: 100%; padding-bottom: 73.93767705382436%;"
          />
          <img
            alt=""
            aria-hidden="true"
            src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAPABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAQFAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAABAv/aAAwDAQACEAMQAAABY4VxKBIGf//EABoQAAIDAQEAAAAAAAAAAAAAAAABAgMREjL/2gAIAQEAAQUCrjpP1qE3lksOz//EABYRAQEBAAAAAAAAAAAAAAAAAAAREv/aAAgBAwEBPwHSv//EABURAQEAAAAAAAAAAAAAAAAAAAAR/9oACAECAQE/AYj/xAAcEAACAQUBAAAAAAAAAAAAAAAAATECEBJBUWH/2gAIAQEABj8C6JUySZ6F7b//xAAcEAACAgIDAAAAAAAAAAAAAAABEQAxIVEQQZH/2gAIAQEAAT8hIHQaigsQF4gXT2NWQVuEdhW3x//aAAwDAQACAAMAAAAQ4+//xAAXEQADAQAAAAAAAAAAAAAAAAAAAREh/9oACAEDAQE/EMVUo//EABYRAQEBAAAAAAAAAAAAAAAAAAARUf/aAAgBAgEBPxDaH//EAB4QAQACAgEFAAAAAAAAAAAAAAERIQAxcUFRYZGx/9oACAEBAAE/EEKI60I86xyRZUEjqPvNJHwy9miG2LXgZRab+9sUtAHGf//Z"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
          />
          <picture>
            <source
              sizes="(max-width: 706px) 100vw, 706px"
              srcset="test-file-stub"
            />
            <img
              alt="Alt Text"
              loading="lazy"
              sizes="(max-width: 706px) 100vw, 706px"
              src="test-file-stub"
              srcset="test-file-stub"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
            />
          </picture>
          <noscript>
            &lt;picture&gt;&lt;source srcset="test-file-stub" sizes="(max-width: 706px) 100vw, 706px" /&gt;&lt;img loading="lazy" sizes="(max-width: 706px) 100vw, 706px" srcset="test-file-stub" src="test-file-stub" alt="Alt Text" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
          </noscript>
        </div>
      </div>
      <div
        class="split-content__content"
      >
        <h2
          class="emotion-2 emotion-3"
        >
          Heading
        </h2>
        <h3
          class="emotion-4 emotion-5"
        >
          Subheading
        </h3>
        <p>
          Main Content
        </p>
      </div>
    </div>
  </section>
</div>
`;
